src/mui/field/ReferenceArrayField.js

```diff
@@ -90,7 +90,7 @@ ReferenceArrayField.propTypes = {
 
 const mapStateToProps = (state, props) => {
     const { record, source, reference } = props;
-    const ids = get(record, source);
+    const ids = get(record, source) || [];
```

Comment:
Each time the state changes, redux will create a new empty array. The underlying component is pure, therefore it compares props by reference. An empty array is a new object, so the component will always rerender. Maybe there are other factors already forcing a rerender (which is why don don't see any change), but it's not a reason to add new ones ;)